MAINT Deprecate `matching` as metric (#26264)
scikit-learn/scikit-learn: doc/whats_new/v1.3.rst

  `tpr=0` and `fpr=0`.
  :pr:`26194` by :user:`Guillaume Lemaitre <glemaitre>`.

> - |Fix| Deprecate `matching` as a metric to be consistent with `scipy.spatial.distance`.
>   :pr:`26264` by :user:`Barata T. Onggo <magnusbarata>`

Reviewer: ```suggestion
- |Fix| `matching` has been deprecated as a metric when SciPy>=1.9 is used
  to be consistent with `scipy.spatial.distance` which does not support it
  anymore.
  :pr:`26264` by :user:`Barata T. Onggo <magnusbarata>`
```

  `tpr=0` and `fpr=0`.
  :pr:`26194` by :user:`Guillaume Lemaitre <glemaitre>`.

> - |Fix| `matching` has been deprecated as a metric when SciPy>=1.9 is used
>   to be consistent with `scipy.spatial.distance` which does not support it
>   anymore.
>   :pr:`26264` by :user:`Barata T. Onggo <magnusbarata>`

Reviewer 2: Nit: Small nit in wording:

```suggestion
- |Fix| The `'matching'` metric has been removed when using SciPy>=1.9
  to be consistent with `scipy.spatial.distance` which does not support
  `'matching'` anymore.
  :pr:`26264` by :user:`Barata T. Onggo <magnusbarata>`
```
